pulsar-broker/src/test/java/org/apache/pulsar/schema/SchemaTest.java

```diff
@@ -556,4 +558,39 @@ public void testDeleteTopicAndSchema() throws Exception {
             }
         }
     }
+
+    @Test
+    public void testProducerMultipleSchemaMessages() throws Exception {
+        final String tenant = PUBLIC_TENANT;
+        final String namespace = "test-namespace-" + randomName(16);
+        final String topicName = "auto_schema_test";
+
+        String ns = tenant + "/" + namespace;
+        admin.namespaces().createNamespace(ns, Sets.newHashSet(CLUSTER_NAME));
+        admin.namespaces().setSchemaCompatibilityStrategy(ns, SchemaCompatibilityStrategy.ALWAYS_COMPATIBLE);
+
+        final String topic = TopicName.get(TopicDomain.persistent.value(), tenant, namespace, topicName).toString();
+        @Cleanup
+        Producer<byte[]> producer = pulsarClient.newProducer(Schema.AUTO_PRODUCE_BYTES())
```

Comment:
why do you need Schema.AUTO_PRODUCE_BYTES() ?

Reply:
Because the test is want to produce messages with different schemas, otherwise we need to create multiple producers.

Reply:
I thought it works even just with 

`  Producer<byte[]> producer = pulsarClient.newProducer()`

because the schema you provide to the producer is not used when you set the Schema per-message

Reply:
Use the AutoProduceBytesSchema in ALWAYS_COMPATIBLE mode, if the message schema is not in the schema cache of the producer, it could be uploaded to the broker and save in schema storage, and all the schemas could be queried. This could verify the schema cache work well or not.

Reply:
@eolivelli Here is the Proposal of the AUTO_PRODUCE schema https://github.com/apache/pulsar/wiki/PIP-43%3A-producer-send-message-with-different-schema, I think you can get more detail from the PIP.